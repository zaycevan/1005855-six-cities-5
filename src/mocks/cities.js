export default [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
