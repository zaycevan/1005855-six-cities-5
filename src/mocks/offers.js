import {HousingType} from "../const";
import {getRandomInteger} from "../utils/common";

const generateReviewsId = () => {
  const randomReviewsId = new Set();
  for (let i = 0; i < getRandomInteger(1, 6); i++) {
    randomReviewsId.add(getRandomInteger(11, 16));
  }

  return Array.from(randomReviewsId);
};

const generateRating = () => {
  const rating = (Math.floor((Math.random() * 5) * 10) / 10);
  return rating;
};

export default [
  {
    id: 1,
    photos: [`https://roomester.ru/wp-content/uploads/2018/08/skandinavskij-stil-v-interere-gostinoj-69.jpg`, `https://roomester.ru/wp-content/uploads/2019/03/gostinaya-spalnya-4-2.jpg`, `https://roomester.ru/wp-content/uploads/2019/02/gostinaya-v-stile-provans.jpg`, `https://roomester.ru/wp-content/uploads/2018/11/gostinaya-v-stile-loft.jpg`],
    title: `Lovely Guesthouse in the suburbs`,
    description: `Quiet and cosy tiny house in the suburbs of Amsterdam, just 10 minutes by metro from the city center of Amsterdam and 5 minutes away from Amsterdam Ajax Arena. The house is just 20 square meters, but it has everything you might need. It is located in a residential neighborhood, 2 minutes away from the metro station in a lovely green area. It is the ideal place for a couple.`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.APARTMENT,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `dishwasher`, `parking`, `washer`, `elevator`, `conditioner`, `Baby seat`],
    coordinates: [52.3909553943508, 4.85309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/pictures/user/9705ae17-cd68-4a1b-8668-46611c1811e7.jpg?im_w=240`,
      name: `Philip`,
      super: true
    },
    reviewsId: generateReviewsId()
  },
  {
    id: 2,
    photos: [`https://a0.muscache.com/im/pictures/fd1fa9b7-3459-4b4b-91ca-4d62772da7b3.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/af653a13-7892-4899-ae64-0481c5407cd5.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/6b95dbde-95a2-404a-814f-ac00845e2c7b.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/e01fd769-1fb4-41b1-8af7-0a435a4069b4.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/a0c674a1-0154-473d-abcc-bf2b432a392a.jpg?im_w=720`],
    title: `Central location private room all facillities`,
    description: `West town center. Private bright room 4x4 m, central heating, fully equipped kitchen, private shower+toilet. Third floor, stairs. WiFi, tv set, DVD, hairdryer. Towels, bed linen provided. Lively lovely neighborhood, quiet and safe at night. Street with characteristic Amsterdam houses and big trees.
    Museums, park, marke`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.ROOM,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `dishwasher`, `Iron`, `TV`, `Washer`],
    coordinates: [52.369553943508, 4.85309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/users/1232843/profile_pic/1317483378/original.jpg?im_w=720`,
      name: `Harm`,
      super: false
    },
    reviewsId: generateReviewsId()
  },
  {
    id: 3,
    photos: [`https://a0.muscache.com/im/pictures/3278447/0cec7b0a_original.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/3278451/c5c62f2c_original.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/3278456/130c852a_original.jpg?im_w=720`],
    title: `Private room in bed and breakfast hosted`,
    description: `Large split level room to acc. 3 people ( possible 4 but then a little crowded but cheaper!!) with fascilities as coffee/tea, radio, fridge, an ajacent bathroom and own toilet (extra toilet owner downstairs) ,microwave, wifi. Typical Amsterdamschool of architecture, good connection ( general transport) with inner city. Nice breakfast and warm hospitality`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.HOUSE,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `Heating`, `Hair dryer`, `Laptop`],
    coordinates: [52.3909553943508, 4.929309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/pictures/user/4af95a33-57b7-4f36-bc28-df74de058cc7.jpg?im_w=720`,
      name: `Erwin`,
      super: false
    },
    reviewsId: generateReviewsId()
  },
  {
    id: 4,
    photos: [`https://a0.muscache.com/im/pictures/12edb6ef-7cef-40ae-a01e-d98ec09d405a.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/b621e711-566e-4fd6-b11e-9a896c3b6aa6.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/1fe51232-1688-4cea-b56e-0ec83f26c0cf.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/0623a3a7-3f6a-435e-a900-f5cc2ec750e0.jpg?im_w=720`],
    title: `Felix Super Apartment`,
    description: `Amazing Apartment in the middel of Amsterdam , only five minutes walking from Centraal Station and five minutes walking from Dam Square .`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.HOTEL,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `Heating`, `TV`],
    coordinates: [52.3809553943508, 4.939309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/pictures/user/2f3dfed9-fe3f-48d7-86dc-acf077d25711.jpg?im_w=720`,
      name: `Dario`,
      super: true
    },
    reviewsId: generateReviewsId()
  }
];
