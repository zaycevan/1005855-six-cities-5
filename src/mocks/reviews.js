import {getRandomInteger} from "@utils/common";

const AVATAR_URL = `https://api.adorable.io/avatars/128`;

const generateRating = () => {
  const rating = (Math.floor((Math.random() * 5) * 10) / 10);
  return rating;
};

const generateReviewDate = () => {
  const currentDate = new Date();
  currentDate.setFullYear(2019, getRandomInteger(0, 11), getRandomInteger(1, 30));
  return new Date(currentDate);
};

export default [
  {
    id: 11,
    avatar: `${AVATAR_URL}/${Math.random()}`,
    name: `Kevin`,
    rating: generateRating(),
    date: generateReviewDate(),
    text: `Cool flat ideal for 4 persons, with a perfect location !! And the owner is also very nice. The only thing is the two stairs because they are a little bit hard but for us it was ok.`
  },
  {
    id: 12,
    avatar: `${AVATAR_URL}/${Math.random()}`,
    name: `Maeva`,
    rating: generateRating(),
    date: generateReviewDate(),
    text: `First of all, the stairs are not good for old people to use. You have to use those stairs carefully. May be, unsure, it's overall in Amsterdam. Please see the picture.`
  },
  {
    id: 13,
    avatar: `${AVATAR_URL}/${Math.random()}`,
    name: `Nataly`,
    rating: generateRating(),
    date: generateReviewDate(),
    text: `Really nice place to stay, reasonably priced, tram is two minutes away with good connection to the city center. Clean rooms and nice staff. Recommended.`
  },
  {
    id: 14,
    avatar: `${AVATAR_URL}/${Math.random()}`,
    name: `Malvika`,
    rating: generateRating(),
    date: generateReviewDate(),
    text: `Really clean and lovely hotel. Had a great time staying here as a group of 6. However the hotel is a bit far from the centre - 45-60 mins via public transport and the taxis in Amsterdam are quite expensive so it ended up costing us a lot as we spent a lot of time in the city.`
  },
  {
    id: 15,
    avatar: `${AVATAR_URL}/${Math.random()}`,
    name: `Hamudi`,
    rating: generateRating(),
    date: generateReviewDate(),
    text: `Great place in a perfect neighborhood. It’s not in the middle of downtown (and doesn’t claim to be) which is the perfect mix of convenient but still quiet.`
  },
  {
    id: 16,
    avatar: `${AVATAR_URL}/${Math.random()}`,
    name: `Dominic`,
    rating: generateRating(),
    date: generateReviewDate(),
    text: `Location and place were fine with exceptions to silk and shower drainage. Check-in was totally irritating and unnecessary to meet with their 3rd party vendor when there was a lock box next to the door. `
  },
];
