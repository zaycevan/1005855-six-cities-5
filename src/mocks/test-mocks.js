import {NameSpace} from "@store/reducers/root-reducer";

export const TestMockData = {
  offer: {
    "city": `Cologne`,
    "cityLocation": [50.938361, 6.959974],
    "previewPhotos": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    "photos": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
    ],
    "title": `The house among olive `,
    "isFavorite": true,
    "isPremium": false,
    "rating": 3.8,
    "type": `house`,
    "bedroomsCount": 3,
    "maxGuestsCount": 4,
    "price": 493,
    "amenities": [
      `Laptop friendly workspace`,
      `Washer`,
      `Towels`
    ],
    "hostInformation": {
      "id": 25,
      "name": `Angelina`,
      "super": true,
      "avatar": `img/avatar-angelina.jpg`
    },
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "coordinates": [50.957361, 6.9509739999999995],
    "id": 1
  },
  offers: [
    {
      "id": 1,
      "city": `Cologne`,
      "cityLocation": [50.938361, 6.959974],
      "previewPhotos": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      "photos": [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
      ],
      "title": `The house among olive `,
      "isFavorite": true,
      "isPremium": false,
      "rating": 3.8,
      "type": `house`,
      "bedroomsCount": 3,
      "maxGuestsCount": 4,
      "price": 493,
      "amenities": [
        `Laptop friendly workspace`,
        `Washer`,
        `Towels`
      ],
      "hostInformation": {
        "id": 25,
        "name": `Angelina`,
        "super": true,
        "avatar": `img/avatar-angelina.jpg`
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "coordinates": [50.957361, 6.9509739999999995]
    }
  ],
  offersFromServer: [
    {
      "bedrooms": 3,
      "city": {
        "location": {
          "latitude": 50.938361,
          "longitude": 6.959974,
          "zoom": 10
        },
        "name": `Cologne`
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "goods": [`Laptop friendly workspace`, `Washer`, `Towels`],
      "host": {
        "avatar_url": `img/avatar-angelina.jpg`,
        "id": 25,
        "is_pro": true,
        "name": `Angelina`
      },
      "id": 1,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      "is_favorite": true,
      "is_premium": false,
      "location": {
        "latitude": 50.957361,
        "longitude": 6.9509739999999995,
        "zoom": 8
      },
      "max_adults": 4,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      "price": 493,
      "rating": 3.8,
      "title": `The house among olive `,
      "type": `house`
    },
  ],
  offerFromServer: {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 50.938361,
        "longitude": 6.959974,
        "zoom": 10
      },
      "name": `Cologne`
    },
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "goods": [`Laptop friendly workspace`, `Washer`, `Towels`],
    "host": {
      "avatar_url": `img/avatar-angelina.jpg`,
      "id": 25,
      "is_pro": true,
      "name": `Angelina`
    },
    "id": 1,
    "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    "is_favorite": false,
    "is_premium": false,
    "location": {
      "latitude": 50.957361,
      "longitude": 6.9509739999999995,
      "zoom": 8
    },
    "max_adults": 4,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    "price": 493,
    "rating": 3.8,
    "title": `The house among olive `,
    "type": `house`
  },
  review: {
    "id": 1,
    "name": `Corey`,
    "avatar": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    "authorId": 1,
    "authorSuper": true,
    "rating": 5,
    "text": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    "date": new Date(`2020-11-01T13:38:44.752Z`)
  },
  reviews: [
    {
      "id": 1,
      "name": `Corey`,
      "avatar": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
      "authorId": 1,
      "authorSuper": true,
      "rating": 5,
      "text": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
      "date": new Date(`2020-11-01T13:38:44.752Z`)
    }
  ],
  reviewsFromServer: [
    {
      "comment": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
      "date": `2020-11-01T13:38:44.752Z`,
      "id": 1,
      "rating": 5,
      "user": {
        "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
        "id": 1,
        "is_pro": true,
        "name": `Corey`
      }
    }
  ],
  user: {
    "email": `Oliver.conner@gmail.com`,
    "password": `12345678`
  },
  appStatus: `APP_READY`,
  currentCity: `Paris`,
  cities: [`Paris`, `Cologne`, `Brussels`],
  type: `Main`,
  currentSort: `POPULAR`
};

export const TestMockStore = {
  [NameSpace.CITIES]: {
    cities: TestMockData.cities,
    currentCity: TestMockData.currentCity,
  },
  [NameSpace.OFFERS]: {
    offers: TestMockData.offers,
    nearOffers: TestMockData.offers,
    favoriteOffers: TestMockData.offers,
    favoriteOfferStatus: `BEFORE_SENT`,
    favoriteOfferId: 1
  },
  [NameSpace.REVIEWS]: {
    reviews: TestMockData.reviews,
    reviewStatus: `BEFORE_SENT`
  },
  [NameSpace.SORT]: {
    currentSort: `POPULAR`,
  },
  [NameSpace.USER]: {
    user: TestMockData.user.email,
    authorizationStatus: `NO_AUTH`,
  },
  [NameSpace.APP]: {
    appStatus: `NOT_READY`,
  }
};
