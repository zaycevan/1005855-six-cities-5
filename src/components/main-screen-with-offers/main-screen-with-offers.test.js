import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import browserHistory from "@src/browser-history";
import configureStore from "redux-mock-store";
import MainScreenWithOffers from "./main-screen-with-offers-connect";
import {TestMockStore} from "@src/mocks/test-mocks";

const noop = () => {};
const mockStore = configureStore([]);
const store = mockStore(TestMockStore);

const tree = renderer.create(
    <Provider store={store}>
      <Router history={browserHistory}>
        <MainScreenWithOffers
          activeCardId={1}
          onActiveCard={noop}
          outActiveCard={noop}
        />
      </Router>
    </Provider>
);

it(`Should MainScreenWithOffers render correctly`, () => {
  expect(tree.toJSON()).toMatchSnapshot();
});
