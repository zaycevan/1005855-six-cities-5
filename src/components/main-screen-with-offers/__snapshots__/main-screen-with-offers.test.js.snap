// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainScreenWithOffers render correctly 1`] = `
<div
  className="cities__places-container container"
>
  <section
    className="cities__places places"
  >
    <h2
      className="visually-hidden"
    >
      Places
    </h2>
    <b
      className="places__found"
    >
      0
       places to stay in 
      Paris
    </b>
    <form
      action="#"
      className="places__sorting"
      method="get"
    >
      <span
        className="places__sorting-caption"
      >
        Sort by
      </span>
      <span
        className="places__sorting-type"
        onClick={[Function]}
        tabIndex="0"
      >
        POPULAR
        <svg
          className="places__sorting-arrow"
          height="4"
          width="7"
        >
          <use
            xlinkHref="#icon-arrow-select"
          />
        </svg>
      </span>
      <ul
        className="places__options places__options--custom "
      >
        <li
          className="places__option "
          onClick={[Function]}
          tabIndex="0"
        >
          Popular
        </li>
        <li
          className="places__option "
          onClick={[Function]}
          tabIndex="0"
        >
          Price: low to high
        </li>
        <li
          className="places__option "
          onClick={[Function]}
          tabIndex="0"
        >
          Price: high to low
        </li>
        <li
          className="places__option "
          onClick={[Function]}
          tabIndex="0"
        >
          Top rated first
        </li>
      </ul>
    </form>
    <div
      className="cities__places-list places__list tabs__content"
    />
  </section>
  <div
    className="cities__right-section"
  >
    <section
      className="cities__map map"
      id="map"
    />
  </div>
</div>
`;
